c3/c3objs.py

```diff
@@ -281,11 +277,13 @@ def _set_value(self, val) -> None:
         else:
             val = tf.constant(val, tf.float64)
 
-        tmp = 2 * (val * self.pref - self.offset) / self.scale - 1
+        tmp = (
+            2 * (tf.reshape(val, self.shape) * self.pref - self.offset) / self.scale - 1
+        )
 
         if np.any(tf.math.abs(tmp) > tf.constant(1.0, tf.float64)):
             raise Exception(
```

Comment:
Top level catch-all `Exception` should be avoided whenever possible. An `IndexError` seems more suitable for this.